src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resArray = [];
+  const newState = {
+    ...state,
+  };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        for (const key in newState) {
+          delete newState[key];
+        }
+        resArray.push({ ...newState });
```

Comment:
p4 of checklist. 
You don't need to push your newState in every case. Just do it once in the end of cycle iteration.